bot/cogs/snekbox.py

```diff
@@ -199,32 +203,64 @@ def get_status_emoji(results: dict) -> str:
 
         log.info(f"Received code from {ctx.author} for evaluation:\n{code}")
 
-        self.jobs[ctx.author.id] = datetime.datetime.now()
-        code = self.prepare_input(code)
+        while True:
+            self.jobs[ctx.author.id] = datetime.datetime.now()
+            code = self.prepare_input(code)
 
-        try:
-            async with ctx.typing():
-                results = await self.post_eval(code)
-                msg, error = self.get_results_message(results)
-
-                if error:
-                    output, paste_link = error, None
-                else:
-                    output, paste_link = await self.format_output(results["stdout"])
-
-                icon = self.get_status_emoji(results)
-                msg = f"{ctx.author.mention} {icon} {msg}.\n\n```py\n{output}\n```"
-                if paste_link:
-                    msg = f"{msg}\nFull output: {paste_link}"
-
-                response = await ctx.send(msg)
-                self.bot.loop.create_task(
-                    wait_for_deletion(response, user_ids=(ctx.author.id,), client=ctx.bot)
+            try:
+                async with ctx.typing():
+                    results = await self.post_eval(code)
+                    msg, error = self.get_results_message(results)
+
+                    if error:
+                        output, paste_link = error, None
+                    else:
+                        output, paste_link = await self.format_output(results["stdout"])
+
+                    icon = self.get_status_emoji(results)
+                    msg = f"{ctx.author.mention} {icon} {msg}.\n\n```py\n{output}\n```"
+                    if paste_link:
+                        msg = f"{msg}\nFull output: {paste_link}"
+
+                    response = await ctx.send(msg)
+                    self.bot.loop.create_task(
+                        wait_for_deletion(response, user_ids=(ctx.author.id,), client=ctx.bot)
+                    )
+
+                    log.info(f"{ctx.author}'s job had a return code of {results['returncode']}")
+            finally:
+                del self.jobs[ctx.author.id]
+
+            try:
+                _, new_message = await self.bot.wait_for(
+                    'message_edit',
+                    check=partial(predicate_eval_message_edit, ctx),
```

Comment:
This will make the bot creating a partial everytime, even when the `ctx` will not be changed.

Suggestions:
Create the partial outside of the loop, and then call it inside like this
```py
_predicate_eval_message_edit = partial(predicate_eval_message_edit, ctx)

...

await self.bot.wait_for(
    'message_edit',
    check=predicate_eval_message_edit,
    timeout=10
)
```

This way, we will not create 2 new partials via every loop even when someone is re-evaluating their code a lot of times.